internal/nostr/nostr.go

```diff
@@ -203,6 +203,59 @@ func (svc *Service) InfoHandler(c echo.Context) error {
 	}
 }
 
+func (svc *Service) PublishHandler(c echo.Context) error {
+	var requestData PublishRequest
+	if err := c.Bind(&requestData); err != nil {
+		return c.JSON(http.StatusBadRequest, ErrorResponse{
+			Message: fmt.Sprintf("error decoding publish request: %s", err.Error()),
+		})
+	}
+
+	if (requestData.SignedEvent == nil) {
+		return c.JSON(http.StatusBadRequest, ErrorResponse{
+			Message: "signed event is empty",
+		})
+	}
+
+	ctx, cancel := context.WithTimeout(c.Request().Context(), 90*time.Second)
+	defer cancel()
+
+	relay, isCustomRelay, err := svc.getRelayConnection(ctx, requestData.RelayUrl)
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, ErrorResponse{
+			Message: "error connecting to relay: " + err.Error(),
+		})
+	}
+
+	if isCustomRelay {
+		defer relay.Close()
+	}
+
+	svc.Logger.WithFields(logrus.Fields{
+		"eventId":   requestData.SignedEvent.ID,
+		"relayUrl":  requestData.RelayUrl,
+	}).Info("Publishing event...")
+
+	err = relay.Publish(ctx, *requestData.SignedEvent)
+	if err != nil {
+		svc.Logger.WithError(err).WithFields(logrus.Fields{
+			"eventId":   requestData.SignedEvent.ID,
+			"relayUrl":  requestData.RelayUrl,
+		}).Error("Failed to publish event")
+	
+		return c.JSON(http.StatusInternalServerError, ErrorResponse{
+			Message: "error publishing the event: " + err.Error(),
+		})
+	}
+
+	svc.Logger.WithFields(logrus.Fields{
+		"eventId":   requestData.SignedEvent.ID,
+		"relayUrl":  requestData.RelayUrl,
+	}).Info("Published event")
+
+	return c.JSON(http.StatusOK, "published")
```

Comment:
be consistent with the API response. 
always return JSON, always try to return something useful.